0.21.0 functional tests failing to compile

The functional tests are failing to compile on the 0.21.0 release for Java 11 and 14.
Example failure https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_ppc64_aix_Release_testList_1/4
```
11:59:29      [javac] /home/****/workspace/Test_openjdk11_j9_sanity.functional_ppc64_aix_Release_testList_1/openjdk-tests/functional/Java11andUp/src/org/openj9/test/condy/CondyGenerator.java:24: error: package org.objectweb.asm does not exist
11:59:29      [javac] import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
11:59:29      [javac]                                ^
```
https://ci.eclipse.org/openj9/job/Pipeline-Release-0.21.0-Build11/11/
https://ci.eclipse.org/openj9/job/Pipeline-Release-0.21.0-Build14/5/

--- a/test/functional/Java11andUp/build.xml
+++ b/test/functional/Java11andUp/build.xml
@@ -56,7 +56,7 @@
 			<include name="**/condy/BootstrapMethods.java" />
 			<include name="**/condy/CondyGenerator.java" />
 			<classpath>
-				<pathelement location="${LIB_DIR}/asm-7.3.jar" />
+				<pathelement location="${LIB_DIR}/asm-8.0.1.jar" />
 			</classpath>
 		</javac>
 	</target>
@@ -67,7 +67,7 @@
 		<java classname="org.openj9.test.condy.CondyGenerator" fork="true" failonerror="true" logError="true" jvm="${javaexecutable.java}">
 			<jvmarg value="-Xverify:none" />
 			<classpath>
-				<pathelement location="${LIB_DIR}/asm-7.3.jar" />
+				<pathelement location="${LIB_DIR}/asm-8.0.1.jar" />
 				<pathelement location="${build}" />
 			</classpath>
 		</java>
@@ -94,7 +94,7 @@
 			<classpath>
 				<pathelement location="${LIB_DIR}/testng.jar"/>
 				<pathelement location="${LIB_DIR}/jcommander.jar"/>
-				<pathelement location="${LIB_DIR}/asm-7.3.jar" />
+				<pathelement location="${LIB_DIR}/asm-8.0.1.jar" />
 				<pathelement location="${build}" />
 			</classpath>
 		</javac>

--- a/test/functional/Java11andUp/playlist.xml
+++ b/test/functional/Java11andUp/playlist.xml
@@ -173,7 +173,7 @@
 			<variation>-Xgcpolicy:balanced</variation>
 		</variations>
 		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) \
-			-cp $(Q)$(LIB_DIR)$(D)asm-7.3.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
+			-cp $(Q)$(LIB_DIR)$(D)asm-8.0.1.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
 			org.testng.TestNG -d $(REPORTDIR) $(Q)$(TEST_RESROOT)$(D)testng.xml$(Q) -testnames GarbageCollectionCondyTest \
 			-groups $(TEST_GROUP) \
 			-excludegroups $(DEFAULT_EXCLUDE); \
@@ -206,7 +206,7 @@
 			<variation>-Xgcpolicy:metronome</variation>
 		</variations>
 		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) \
-			-cp $(Q)$(LIB_DIR)$(D)asm-7.3.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
+			-cp $(Q)$(LIB_DIR)$(D)asm-8.0.1.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
 			org.testng.TestNG -d $(REPORTDIR) $(Q)$(TEST_RESROOT)$(D)testng.xml$(Q) -testnames GarbageCollectionCondyTest \
 			-groups $(TEST_GROUP) \
 			-excludegroups $(DEFAULT_EXCLUDE); \
@@ -238,7 +238,7 @@
 			<variation>-Xgcpolicy:metronome</variation>
 		</variations>
 		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) \
-			-cp $(Q)$(LIB_DIR)$(D)asm-7.3.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
+			-cp $(Q)$(LIB_DIR)$(D)asm-8.0.1.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
 			org.testng.TestNG -d $(REPORTDIR) $(Q)$(TEST_RESROOT)$(D)testng.xml$(Q) -testnames GarbageCollectionCondyTest \
 			-groups $(TEST_GROUP) \
 			-excludegroups $(DEFAULT_EXCLUDE); \

--- a/test/functional/Java14andUp/build.xml
+++ b/test/functional/Java14andUp/build.xml
@@ -55,7 +55,7 @@
 			<classpath>
 				<pathelement location="${LIB_DIR}/testng.jar"/>
 				<pathelement location="${LIB_DIR}/jcommander.jar"/>
-				<pathelement location="${LIB_DIR}/asm-7.3.jar" />
+				<pathelement location="${LIB_DIR}/asm-8.0.1.jar" />
 				<pathelement location="${build}" />
 			</classpath>
 		</javac>

--- a/test/functional/Java14andUp/playlist.xml
+++ b/test/functional/Java14andUp/playlist.xml
@@ -29,7 +29,7 @@
             <variation>--enable-preview</variation>
         </variations>
         <command>$(JAVA_COMMAND) $(JVM_OPTIONS) \
-            -cp $(Q)$(LIB_DIR)$(D)asm-7.3.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
+            -cp $(Q)$(LIB_DIR)$(D)asm-8.0.1.jar$(P)$(RESOURCES_DIR)$(P)$(TESTNG)$(P)$(TEST_RESROOT)$(D)GeneralTest.jar$(Q) \
             org.testng.TestNG -d $(REPORTDIR) $(Q)$(TEST_RESROOT)$(D)testng.xml$(Q) -testnames Jep359Tests \
             -groups $(TEST_GROUP) \
             -excludegroups $(DEFAULT_EXCLUDE); \

--- a/test/functional/Java14andUp/src/org/openj9/test/utilities/RecordClassGenerator.java
+++ b/test/functional/Java14andUp/src/org/openj9/test/utilities/RecordClassGenerator.java
@@ -32,14 +32,13 @@ import org.objectweb.asm.*;
         cw.visit(V14 | V_PREVIEW, ACC_FINAL | ACC_SUPER, className, null, "java/lang/Record", null);
 
         /* add record component */
-        RecordComponentVisitor rcv = cw.visitRecordComponentExperimental(
-                ACC_DEPRECATED,
+        RecordComponentVisitor rcv = cw.visitRecordComponent(
                 rcName,
                 rcType,
                 rcSignature
             );
 
-        rcv.visitEndExperimental();
+        rcv.visitEnd();
 
         /* add accessor method for record component */
         MethodVisitor mv = cw.visitMethod(ACC_PUBLIC, rcName, "()" + rcType, null, null);
@@ -69,14 +68,13 @@ import org.objectweb.asm.*;
         ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_FRAMES);
         cw.visit(V14 | V_PREVIEW, ACC_FINAL | ACC_SUPER, className, null, "java/lang/Record", null);
 
-        RecordComponentVisitor rcv = cw.visitRecordComponentExperimental(
-                ACC_DEPRECATED,
+        RecordComponentVisitor rcv = cw.visitRecordComponent(
                 rcName,
                 rcType,
                 null
             );
 
-        rcv.visitEndExperimental();
+        rcv.visitEnd();
 
         cw.visitEnd();
         return cw.toByteArray();
@@ -86,14 +84,13 @@ import org.objectweb.asm.*;
         ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_FRAMES);
         cw.visit(V14 | V_PREVIEW, ACC_FINAL | ACC_SUPER, className, null, "java/lang/Record", null);
 
-        RecordComponentVisitor rcv = cw.visitRecordComponentExperimental(
-                ACC_DEPRECATED,
+        RecordComponentVisitor rcv = cw.visitRecordComponent(
                 rcName,
                 rcType,
                 null
             );
 
-        rcv.visitEndExperimental();
+        rcv.visitEnd();
 
         /* add invalid method for record component */
         MethodVisitor mv = cw.visitMethod(ACC_PUBLIC, rcName, "()" + invalidType, null, null);

--- a/test/functional/cmdLineTests/lockWordAlignment/alignmentSettings.mk
+++ b/test/functional/cmdLineTests/lockWordAlignment/alignmentSettings.mk
@@ -26,5 +26,5 @@ ASM_JAR=$(LIB_DIR)$(D)asm-all.jar
 # if JDK_VERSION is not 8
 ifeq ($(filter 8, $(JDK_VERSION)),)
  ADD_EXPORTS=--add-exports=java.base/com.ibm.oti.vm=ALL-UNNAMED
- ASM_JAR=$(LIB_DIR)$(D)asm-7.3.jar
+ ASM_JAR=$(LIB_DIR)$(D)asm-8.0.1.jar
 endif

